curp/src/server/bg_tasks.rs

```diff
@@ -125,6 +118,73 @@ pub(super) async fn run_bg_tasks<
     info!("all background task stopped");
 }
 
+/// Do periodical task
+#[allow(clippy::integer_arithmetic)] // can't overflow
+async fn bg_tick<C: Command + 'static, Conn: ConnectInterface, ExeTx: CmdExeSenderInterface<C>>(
+    connects: Vec<Arc<Conn>>,
+    state: StateRef<C, ExeTx>,
+    timeout: Arc<ServerTimeout>,
+) {
+    let (rpc_timeout, heartbeat_interval, follower_timeout_ticks, candidate_timeout_ticks) = (
+        *timeout.rpc_timeout(),
+        *timeout.heartbeat_interval(),
+        *timeout.follower_timeout_ticks(),
+        *timeout.candidate_timeout_ticks(),
+    );
+    // randomize to minimize vote split possibility
+    let follower_timeout_ticks =
+        follower_timeout_ticks + thread_rng().gen_range(0..follower_timeout_ticks);
+    let candidate_timeout_ticks =
```

Comment:
If we really need to separate the leader failure detection and election timeouts into two parts, the `follower_timeout_ticks` and the `candidate_timeout_ticks`, for better performance, then the `candidate_timeout_ticks` should be randomized every time when split vote happens.

Reply:
I read some more of etcd's code. Etcd's raft resets its timeout ticks every time it update's its term. I think it's a better option since we don't have to generate some random number every tick and can also avoid electing forever.